### DISCUSSION TITLE
IoC Module Reuse

### DISCUSSION THREAD

**Discussion by sigmarsson** [canton7/Stylet#229] [Ideas]
Hi,

Is it possible to reuse a pre-configured IoC module in Stylet's bootstrapper ?

`protected override void ConfigureIoC(IStyletIoCBuilder builder);`

NInject and DryIoc both maintains aspiration to keep configurations in a dedicated module, those modules facilitate unit testing as well. 

### NInject

```
public class MyIoCSetup : NinjectModule
{
  public override void Load()
  {
     // ... Perform bindings here. 
  }
}
```

### DryIoC

https://github.com/dadhi/DryIoc/issues/326

**Comment by canton7** [comment]
https://github.com/canton7/Stylet/wiki/StyletIoC-Modules ?

**Comment by sigmarsson** [comment]
That is cool. Do you think it would be possible to standardize these modules across the globe by having a `IIoCModule` interface or alike?

Basically, the mapping might be very similar duty for all IoC containers. I am asking all this because you earlier stipulated that you are not keen on delivering a separate nuget package for a stand alone Stylet IoC. I have a configuration set up for a different infrastructure now with NInject and wish transfer and apply that config. DryIoC will offer this soon perhaps with the next feat release. Do you see a chance to  fusion with others on any degree on this matter ? Thanks !

![image](https://user-images.githubusercontent.com/43340959/110814844-20a95c80-828a-11eb-91e2-6545668a95d5.png)

![image](https://user-images.githubusercontent.com/43340959/110814913-3585f000-828a-11eb-8fae-8830ca199f76.png)

**Reply by canton7** [reply]
I don't follow I'm afraid. Every ioc container has a different interface for registering types with it. That `Bind` call in `StyletIoCModule` is only the start: there are a ton of other methods you call on the `IBindTo` which is returned.

If there's ever going to be standardisation, it will probably be everyone moving to `Microsoft.Extensions.DependencyInjection`.

If you've already got infrastructure set up with Ninject, then just [use the `NinjectBootstrapper`](https://github.com/canton7/Stylet/wiki/Bootstrapper#using-a-custom-ioc-container) and drop StyletIoC entirely.
